feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-page-layout-handler.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { Type } from '@angular/core';
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import {
+  ConfiguratorModelUtils,
+  ConfiguratorRouter,
+  ConfiguratorRouterExtractorService,
+} from '@spartacus/product-configurator/common';
+import { cold } from 'jasmine-marbles';
+import { Observable, of } from 'rxjs';
+
+import { CpqConfiguratorPageLayoutHandler } from './cpq-configurator-page-layout-handler';
+
+const standardRouterData: ConfiguratorRouter.Data = {
+  owner: ConfiguratorModelUtils.createInitialOwner(),
+};
+let routerData: ConfiguratorRouter.Data;
+class MockRouterExtractorService {
+  extractRouterData(): Observable<ConfiguratorRouter.Data> {
+    return of(routerData);
+  }
+}
+const headerSlots = ['SiteLogo', 'MiniCart'];
+const headerSlotsIncludingPreHeader = ['PreHeader', 'SiteLogo', 'MiniCart'];
+const contentSlots = [
+  'CpqConfigHeader',
+  'CpqConfigBanner',
+  'CpqConfigMenu',
+  'CpqConfigContent',
+  'CpqConfigOverviewBanner',
+  'CpqConfigOverviewContent',
+  'CpqConfigBottombar',
+];
+const pageTemplateCpq = 'CpqConfigurationTemplate';
+const pageTemplateOther = 'OtherTemplate';
+const sectionHeader = 'header';
+const sectionContent = 'content';
+
+describe('CpqConfiguratorPageLayoutHandler', () => {
+  let serviceUnderTest: CpqConfiguratorPageLayoutHandler;
```

Comment:
Is it not better to name `classUnderTest` or `handlerUnderTest`? What do you think?

Reply:
Yes, agree, changed that